File: tech.stack.packages/ASP.NET Core/exception/ProcessingException.cs
#header()

#set( $appName = $aib.getApplicationNameFormatted() )
#set( $className = $className )

/**
 * Used to indicate an error occurred in generic processing.
 * <p>
 * @author ${aib.getAuthor()}
 */

namespace ${appName}.Exceptions
{
    /// <summary>
    /// Common  exception thrown during the processing a business request
    /// </summary>
	public class ProcessingException: System.Exception
	{
        /// <summary>
        /// sole constructor
        /// </summary>
        /// <param name="message"></param>
   		public ProcessingException(string message): base(message)
   		{
   		}
	}
}

File: tech.stack.packages/Spring MongoDB/exception/ProcessingException.java
#header()
package ${aib.getRootPackageName()}.exception;

//***********************************
// Imports
//***********************************

/**
 * Used to indicate an error occured in generic processing.
 * <p>
 * @author ${aib.getAuthor()}
 */
public class ProcessingException extends Exception
{

//************************************************************************    
// Public Methods
//************************************************************************

    /** 
     * Base constructor.
     */
    public ProcessingException()
    {
        super(); 
    }

    /** Constructor with message.
     * @param message text of the exception
     */
    public ProcessingException( String message )
    {
        super( message ); 
    }

    /**
     * Constructor with a Throwabe for chained exception and a message.
     * 
     * @param message
     * @param exception
     */
    public ProcessingException( String message, Throwable exception )
    {
        super( message ); 
    }

//************************************************************************    
// Private / Protected Methods
//************************************************************************

//************************************************************************    
// Attributes
//************************************************************************
}




